### PR TITLE
Fix/pre testnet bugfixes and optimizations (#59)

### PR DESCRIPTION
**Pull request by gene-zhan**
* test: add log

* test: setup remote debug for local devnet

* fix: try fix aspect state diverge issue

* fix: native contract bug

* fix: native contract bug

* fix: aspect store error

* fix: unbind error

* fix: unbind error

* fix: unbind error

* fix: unbind error for add log

* fix: unbind error for add log

* fix: unbind error for add log

* fix: unbind error

* fix: delete  log

* Test/remote debug native (#51)

* fix: native contract bug

* fix: native contract bug

* fix: aspect store error

* fix: unbind error

* fix: unbind error

* fix: unbind error for add log

* fix: unbind error for add log

* fix: unbind error for add log

* fix: unbind error

* fix: delete  log

* fix: update cmd root reference.

* fix: add join point

* fix: upgrade

* fix: fixed jit call return error

* optimize: separate system error and jit call revert error

* feat: check if the join point has run permissions

* feat: check if the join point has run permissions

* fix:

1. rename ContractsOf to BoundAddressesOf
2. fix something

* fix:

1. rename ContractsOf to BoundAddressesOf
2. fix something

* getAspectCode

---------

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ ldflags = -X $(REPO)/version.AppVersion=$(VERSION) \
           -X $(REPO)/version.GitCommitDate=$(GIT_COMMIT_DATE)
 
 ifeq (,$(findstring nostrip,$(COSMOS_BUILD_OPTIONS)))
- ldflags += -w -s
+#  ldflags += -w -s
```
Comment (dumbeng) [inline Makefile:19]: Remove this

File: x/evm/artela/contract/service.go [hunk 2]
```diff
@@ -194,3 +243,17 @@ func (service *AspectService) GetAspectProof(ctx sdk.Context, aspectId common.Ad
 func (service *AspectService) GetBlockHeight() int64 {
 	return service.getHeight()
 }
+
+func (service *AspectService) GetAspectJoinPoint(ctx sdk.Context, aspectId common.Address, version *uint256.Int, commit bool) *big.Int {
+	if commit {
+		sdkCtx, getErr := service.getCtxByHeight(ctx.BlockHeight()-1, true)
```
Comment (dumbeng) [inline x/evm/artela/contract/service.go:249]: I remembered that we have discussed that we will not use block - 1 anymore. You already have this sdkCtx as your first input param, why using getCtxByHeight again? This code makes no sense.

File: x/evm/artela/contract/store.go [hunk 3]
```diff
@@ -167,6 +210,20 @@ func (k *AspectStore) StoreAspectProperty(ctx sdk.Context, aspectId common.Addre
...
+	keyAry := make([]string, keySet.Size())
+	for i, key := range keySet.Values() {
+		keyAry[i] = key.(string)
+	}
+	join := strings.Join(keyAry, types.AspectPropertyAllKeySplit)
+	aspectPropertyKey := types.AspectPropertyKey(
+		aspectId.Bytes(),
+		[]byte(types.AspectPropertyAllKeyPrefix),
+	)
+	aspectPropertyStore.Set(aspectPropertyKey, []byte(join))
```
Comment (dumbeng) [inline x/evm/artela/contract/store.go:224]: rename this store, this should not be called as aspect property store any more, it saves too many other stuffs

File: x/evm/artela/contract/store.go [hunk 4]
```diff
@@ -167,6 +210,20 @@ func (k *AspectStore) StoreAspectProperty(ctx sdk.Context, aspectId common.Addre
...
 		aspectPropertyStore.Set(aspectPropertyKey, []byte(value))
 	}
+
+	// store AspectPropertyAllKey
+	keyAry := make([]string, keySet.Size())
+	for i, key := range keySet.Values() {
+		keyAry[i] = key.(string)
+	}
+	join := strings.Join(keyAry, types.AspectPropertyAllKeySplit)
+	aspectPropertyKey := types.AspectPropertyKey(
```
Comment (dumbeng) [inline x/evm/artela/contract/store.go:220]: also rename the variable here

File: x/evm/artela/provider/artela.go [hunk 5]
```diff
@@ -132,12 +132,12 @@ func (ArtelaProvider) FilterAspectTx(tx sdk.Msg) bool {
...
 }
 
-func (j *ArtelaProvider) GetTxBondAspects(blockNum int64, address common.Address) ([]*asptypes.AspectCode, error) {
+func (j *ArtelaProvider) GetTxBondAspects(blockNum int64, address common.Address, point asptypes.PointCut) ([]*asptypes.AspectCode, error) {
 	heightCtx, err := j.getCtxByHeight(blockNum-1, true)
 	if err != nil {
 		return nil, err
 	}
-	return j.service.GetAspectForAddr(heightCtx, address, false)
+	return j.service.GetPointsAspect(heightCtx, address, point, false)
```
Comment (dumbeng) [inline x/evm/artela/provider/artela.go:140]: func name is confused, rename is to `GetAspectsForJoinPoint`

File: x/evm/artela/types/aspect_store_key.go [hunk 6]
```diff
@@ -24,16 +24,21 @@ const (
...
-	AspectStateEndBlock       = "State/EndBlock"
+	AspectStateBeginBlock      = "State/BeginBlock"
+	AspectStateDeliverTxState  = "State/DeliverTx"
+	AspectStateEndBlock        = "State/EndBlock"
+	AspectAccountKey           = "Aspect_@Acount@_"
+	AspectProofKey             = "Aspect_@Proof@_"
+	AspectRunJoinPointKey      = "Aspect_@Run@JoinPoint@_"
+	AspectPropertyAllKeyPrefix = "Aspect_@Property@AllKey@_"
+	AspectPropertyAllKeySplit  = "^^^"
+	AspectPropertyLimit        = 256
```
Comment (dumbeng) [inline x/evm/artela/types/aspect_store_key.go:41]: use `math.MaxUint8`

**Review by dumbeng — CHANGES_REQUESTED** [review]

**Review by luketheart — DISMISSED** [review]
LGTM.

File: x/evm/artela/contract/service.go (hunk 2 above)
Comment (gene-zhan) [reply x/evm/artela/contract/service.go:250]: this issue will be resolved in a separate branch

**Review by dumbeng — APPROVED** [review]
Still a few typos, but merge it for now, I'll fix them later

**Review by luketheart — APPROVED** [review]

**Pull request merged**